programs/dynamic-bonding-curve/src/instructions/migration/ix_withdraw_migration_fee.rs

```diff
@@ -55,47 +76,42 @@ pub fn handle_withdraw_migration_fee(ctx: Context<WithdrawMigrationFeeCtx>) -> R
         pool.is_curve_complete(config.migration_quote_threshold),
         PoolError::NotPermitToDoThisAction
     );
-
-    let is_partner = ctx.accounts.sender.key() == config.fee_claimer;
-    let is_creator = ctx.accounts.sender.key() == pool.creator;
-
     let MigrationFeeDistribution {
         creator_migration_fee,
         partner_migration_fee,
     } = config.get_migration_fee_distribution()?;
 
-    let partner_migration_fee = if is_partner {
+    let sender_flag = SenderFlag::try_from(flag).map_err(|_| PoolError::TypeCastFailed)?;
+    let fee = if sender_flag == SenderFlag::Partner {
+        require!(
+            ctx.accounts.sender.key() == config.fee_claimer,
+            PoolError::NotPermitToDoThisAction
+        );
         let mask = PARTNER_MASK;
         // Ensure the partner has never been withdrawn
         require!(
             pool.eligible_to_withdraw_migration_fee(mask),
             PoolError::MigrationFeeHasBeenWithdraw
         );
-        // update creator withdraw migration fee
+        // update partner withdraw migration fee
         pool.update_withdraw_migration_fee(mask);
-
         partner_migration_fee
     } else {
-        0
-    };
-
-    let creator_migration_fee = if is_creator {
+        require!(
+            ctx.accounts.sender.key() == pool.creator,
+            PoolError::NotPermitToDoThisAction
+        );
         let mask = CREATOR_MASK;
-        // Ensure the creator has never been withdrawn
+        // Ensure the partner has never been withdrawn
```

Comment:
Ensure the ~partner~ creator has never been withdrawn

Reply:
fixed